pennylane/compiler/qjit_api.py

```diff
@@ -568,3 +607,28 @@ def __call__(self, *init_state):
             args = fn_res if len(args) > 1 else (fn_res,) if len(args) == 1 else ()
 
         return fn_res
+
+    def _call_capture_enabled(self, *init_state):
+
+        import jax  # pylint: disable=import-outside-toplevel
+
+        for_loop_prim = _get_for_loop_qfunc_prim()
+
+        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)(0, *init_state)
+        jaxpr_consts = jax.numpy.array(jaxpr_body_fn.consts)
```

Comment:
I'm a little confused by this line.  What if the consts are different shapes and types? 

```
x = jax.numpy.array(0.1)
y = jax.numpy.array([1, 2])

@for_loop(0,4,1)
def f(i, a, b):
    return a+x, b+y

f(0.3, jax.numpy.array([5,6]) )
```

I might need to research the effect of casting to an array on jaxpr, but in addition to the shapes/ dtype concern, I also want to be careful about introducing unnecessary copies, as that can lead to excess overhead.

Reply:
Thanks, please note that this is not the final implementation and there are several improvements/changes/tests required (the PR is still a draft)